Karma/Controllers/CharitiesController.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using Karma.Data;
+using Karma.Models;
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+
+namespace Karma.Controllers
+{
+    public class CharitiesController : Controller
+    {
+        private readonly KarmaContext _context;
+
+        private readonly IWebHostEnvironment _iWebHostEnv;
+
+        // Passes an object of type IWebHostEnvironment that carries information about our host environment.
+        public CharitiesController(KarmaContext context, IWebHostEnvironment webHostEnvironment)
+        {
+            _context = context;
+            _iWebHostEnv = webHostEnvironment;
+        }
+
+        // GET: Charities
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.Charity.ToListAsync());
+        }
+        public async Task<IActionResult> FilteredCharities(Enums.Category itemType)
+        {
+            List<Charity> charities = await _context.Charity.ToListAsync();
+
+            List<Charity> filtered = new List<Charity>();
+
+            ViewBag.ItemType = itemType.ToString();
+
+            foreach (Charity c in charities)
+            {
+                c.LoadItemTypes();
+                if (c.ItemTypes.Contains(itemType))
```

Comment:
What happens if there is no charity of selected type, does the web page just throw an exception and break?

Reply:
If there are no charities in the list, an empty model is given to the view, which in return informs the user that there are no charities which are connected with the given item category.

Reply:
I got an exception, too. This happened after pressing the Search button.

![MicrosoftTeams-image (2)](https://user-images.githubusercontent.com/73688133/135969717-b7c0e0d8-133d-4d41-81eb-9c839f7c1847.png)

Reply:
You should try deleting all charities and creating new ones, since it might be a case, that you don't have some charity (that are still on db) files on your local machine.

But I will add a guard in the future to prevent this if somehow files are lost.

Reply:
@ArmintasP 

I think you didn't commit and push the folders that you created. These ones: Charity, Charity/Address, Charity/ItemTypes, wwwroot/CharityImages and  wwwroot/PostImages.

Could you please push them to git?

Because they're listed in Startup.cs, but we do not have them.